File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* dummy = new ListNode(-1);
        dummy->next=head;
        ListNode* ans=dummy;
        ListNode* curr=head;
        while(curr){
            if(curr && curr->next && curr->val==curr->next->val){
                while(curr && curr->next && curr->val==curr->next->val){
                    curr=curr->next;
                }
                dummy->next=curr;
                dummy=dummy->next;
            }
            else{
                dummy=dummy->next;
            }
            curr=curr->next;
        }
        return ans->next;
    }
};

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if(not head):
            return head

        temp=head
        curr=head

        while(curr and curr.next):
            while(curr and curr.next and curr.val==curr.next.val):
                curr=curr.next
            temp.next=curr.next
            temp=temp.next
            curr=curr.next
        return head